### DISCUSSION TITLE
[Other] Any chance I can retreive my accidentally nuked documents?

### DISCUSSION THREAD

**Discussion by AlexKalopsia** [jonaswinkler/paperless-ng#684] [Support]
I would have posted this in the main paperless fork, but given this now became the main one I am doing it here. So, I've done an oopsie and `rm -r` my entire paperless folder, which container the mounted paperless volumes for my Docker container.

I don't know the inner workings of paperless, but since my containers are still running (and I backed them up now - still after the deletion), I am wondering if paperless somehow stores the scanned data somewhere that is not exposed to the mounted folders. My assumption is that everything might go in a db that also got nuked, but I am just checking, maybe some of you can help me out :D 

Thanks

**Comment by jonaswinkler** [comment]
Paperless only stores documents in one place. With the Docker image, `/usr/src/paperless/media` contains the documents, and thumbnails. `/usr/src/paperless/data` has the database (which only contains metadata, not the document files), and other stuff.

The actual location of these probably depends on your system config, but they're usually somewhere around `/var/lib/docker/volumes/`.

If you deleted that, and don't have a backup, well... I'm sorry, nothing I can do :(

There are tools out there that restore deleted files, but they most likely won't be able to restore everything perfectly, especially if the drive has been written to after the oopsie.

I've got the paperless volumes on a file system that supports snapshots; If I ever do something stupid, I can simply revert to a known-good snapshot, and these are backed up elsewhere as well.

**Comment by AlexKalopsia** [comment]
Alright, that's what I thought. Well, it's on me, lesson learned the hard way I guess. Thanks for the quick reply.

**Comment by jonaswinkler** [comment]
If you don't have your physical documents anymore, it's probably best to unplug your drive and ask some professional data recovery experts to do something. It's expensive though, depends on how much you need the documents back.
